### PR TITLE
refactor: replace `#[snafu(backtrace)]` with `Location`

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

replace `#[snafu(backtrace)]` with `Location` in all places except in frontend. Theoretically, location is also enough for external error because we don't care about the internal stack trace of dependencies.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/script/src/python/error.rs [hunk 1]
```diff
@@ -105,24 +105,24 @@ pub enum Error {
...
     #[snafu(display("Failed to create tokio task, source: {}", source))]
     TokioJoin { source: tokio::task::JoinError },
 }
 
-impl From<QueryError> for Error {
-    fn from(source: QueryError) -> Self {
-        Self::DatabaseQuery { source }
-    }
-}
+// impl From<QueryError> for Error {
```
Comment (etolbakov) [inline src/script/src/python/error.rs:121]: nit: possibly a leftover, should it be removed?

File: src/script/src/python/error.rs [hunk 2]
```diff
@@ -151,11 +151,11 @@ impl ErrorExt for Error {
     }
 }
 // impl from for those error so one can use question mark and implicitly cast into `CoprError`
-impl From<DataTypeError> for Error {
-    fn from(e: DataTypeError) -> Self {
-        Self::TypeCast { source: e }
-    }
-}
+// impl From<DataTypeError> for Error {
```
Comment (etolbakov) [inline src/script/src/python/error.rs:154]: nit: possibly a leftover, should it be removed?

File: src/servers/src/error.rs [hunk 3]
```diff
@@ -409,11 +409,11 @@ impl From<std::io::Error> for Error {
     }
 }
 
-impl From<auth::Error> for Error {
-    fn from(e: auth::Error) -> Self {
-        Error::Auth { source: e }
-    }
-}
+// impl From<auth::Error> for Error {
```
Comment (etolbakov) [inline src/servers/src/error.rs:412]: nit: possibly a leftover, should it be removed?

**Review by etolbakov — APPROVED** [review]

File: src/catalog/src/error.rs [hunk 4]
```diff
@@ -32,18 +32,18 @@ pub enum Error {
...
     ))]
     CompileScriptInternal {
-        #[snafu(backtrace)]
+        location: Location,
         source: BoxedError,
     },
     #[snafu(display("Failed to open system catalog table, source: {}", source))]
     OpenSystemCatalog {
-        #[snafu(backtrace)]
+        location: Location,
```
[the remark this answers is not in the document]
Comment (waynexia) [reply src/catalog/src/error.rs:40]: There are two things left, (1) where the location should be displayed. I prefer to place it before source error, (2) I haven't find a way to add location for all. I'm afraid we have to do it manually. Thus this PR only removes all backtraces

**Review by evenyag — APPROVED** [review]
> There are two things left, (1) where the location should be displayed. I prefer to place it before source error, (2) I haven't find a way to add location for all. I'm afraid we have to do it manually. Thus this PR only removes all backtraces


Could you please track these things in an issue @waynexia

**Pull request merged**